### PR TITLE
Added AppStream metadata

### PR DESCRIPTION
**Pull request by Mailaender**
https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html makes you more visible in Linux distributions. I added an https://hughsie.github.io/oars/ rating as well.

### CODE REVIEW

File: misc/starfighter.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    <content_attribute id="violence-fantasy">none</content_attribute>
+    <content_attribute id="violence-realistic">moderate</content_attribute>
+    <content_attribute id="violence-bloodshed">none</content_attribute>
+    <content_attribute id="violence-sexual">none</content_attribute>
+    <content_attribute id="drugs-alcohol">none</content_attribute>
+    <content_attribute id="drugs-narcotics">none</content_attribute>
+    <content_attribute id="drugs-tobacco">none</content_attribute>
+    <content_attribute id="sex-nudity">none</content_attribute>
+    <content_attribute id="sex-themes">none</content_attribute>
+    <content_attribute id="language-profanity">none</content_attribute>
```
Comment (onpon4) [inline misc/starfighter.appdata.xml:47]: I believe this should be "moderate", so I'll be changing it to that.

🕵️

**Review by onpon4 — APPROVED** [review]
Other than the profanity categorization (I believe that should be "moderate"), everything here looks good. Thanks!

🕵️

File: misc/starfighter.appdata.xml (hunk 1 above)
Comment (Mailaender) [reply misc/starfighter.appdata.xml:47]: Do it. I have to admit that I eyeballed those. Just like a real rating guy would do. ;)

Comment (onpon4) [reply misc/starfighter.appdata.xml:47]: Already done. :slightly_smiling_face: I did it right after I said that. Also found some other improvements to make.

🕵️

**Comment by Mailaender** [comment]
Can you maybe tag a new version? I tried to apply your patches including your new app ID, but it is spread out to so many commits.

**Comment by onpon4** [comment]
I don't feel ready at the moment to push out a new Starfighter release because I still need to finish the new soundtrack I'm working on (need to compose new credits, menu, and cutscene music). That said, is there something in particular you need updated? I suppose I could do a 2.4.0-1 tag or something with just the commits you need.

🦇

**Comment by Mailaender** [comment]
I mostly care about metainfo and desktop files as I plan to create a @flathub package. Had trouble picking all the right commits and applying patches in correct order.

**Comment by onpon4** [comment]
Alright, I've pushed a tag that (if I didn't miss anything) has all the commits connected to those things applied:

https://github.com/pr-starfighter/starfighter/releases/tag/v2.4.0-1

Hopefully that suffices, if it doesn't please let me know what else I can do to help.

🦇

**Comment by Mailaender** [comment]
Thanks! Alright, there you go: https://github.com/flathub/flathub/pull/3399

**Pull request merged**